### PR TITLE
Clean up the nino34 index task

### PR DESCRIPTION
**Pull request by xylar**
This clean up is needed to generalize the task in preparation for supporting comparison with a reference run (which will replace one of the panels).

A new task has been added to extract time series needed by indices (which may have different time bounds than time series).  Support for separate time bounds was eliminated, perhaps accidentally, in  #271.

Nino34 spectra are now packed into dictionaries for easier iteration and panels are plotted in loops.

The function for determining the maximum value of all spectra (to determine the bounds of the y axes of these plots) has been cleaned up so it no longer relies on a plot already having been performed). 

All warnings previously produced with `warnings.warn` have been switched to just using `print('Warning:..')` because this produces much more intuitive output without a (nearly always useless) stack trace.

### CODE REVIEW

**Comment by xylar** [comment]
@vanroekel, if you could look over these changes and make sure that you are okay with them, I would appreciate it.

**Comment by xylar** [comment]
@vanroekel and @milenaveneziani, I am not sure we really discussed the decision to have IndexNino34 use the output from MpasTimeSeriesTask (meaning it has the same bounds as all other time series).  This could be reconsidered, or a separate instance of MpasTimeSeriesTask could be used to extract time series for indices (with potentially different bounds).  The purpose of removing the `[index]` section here was to make the config files consistent with the current implementation.  An alternative would be to make the code consistent with the config files instead.

**Comment by xylar** [comment]
## Testing

I tested this with the QU240 test case on my laptop.  Results are nearly identical with only some small variation in the y axis of the spectra and the format of the titles and legends.

**Comment by vanroekel** [comment]
@xylar I think we should have separate bounds for NINO34.  As an example, often we have found a need to ignore the first 10 years or so for a NINO computation, but want to retain years 1-10 for other time series.

**Comment by xylar** [comment]
@vanroekel, okay, sounds good.  I just switched to having separate bounds for indices again.  It didn't take long to implement.

File: mpas_analysis/ocean/meridional_heat_transport.py [hunk 1]
```diff
@@ -6,15 +6,12 @@
...
 import netCDF4
 import os
-import warnings
 
 from ..shared.plot.plotting import plot_vertical_section,\
     setup_colormap, plot_1D
 
 from ..shared.io.utility import build_config_full_path
 
-from ..shared.timekeeping.utility import get_simulation_start_time
```
Comment (xylar) [inline mpas_analysis/ocean/meridional_heat_transport.py]: wasn't being used

File: mpas_analysis/shared/plot/plotting.py [hunk 2]
```diff
@@ -1144,42 +1144,6 @@ def setup_colormap(config, configSectionName, suffix=''):
     return (colormap, colorbarLevels)
 
 
-def plot_size_y_axis(plt, xaxisValues, **data):
```
Comment (xylar) [inline mpas_analysis/shared/plot/plotting.py]: moved to IndexNino34 because it's not likely to be used elsewhere.

**Comment by xylar** [comment]
Great, thanks @vanroekel for the quick review.

**Comment by xylar** [comment]
I'm going to go ahead an merge.

**Pull request merged**